lib/Sema/MiscDiagnostics.cpp

```diff
@@ -436,10 +436,10 @@ static void diagSyntacticUseRestrictions(const Expr *E, const DeclContext *DC,
       auto calleeDefaultArg = getMagicIdentifierDefaultArgKind(calleeParam);
       auto callerDefaultArg = getMagicIdentifierDefaultArgKind(callerParam);
 
-      // If one of the parameters doesn't have a default arg, or they both have
-      // the same one, everything's fine.
+      // If one of the parameters doesn't have a default arg, or they're both
+      // compatible, everything's fine.
       if (!calleeDefaultArg || !callerDefaultArg ||
-          *calleeDefaultArg == *callerDefaultArg)
+          areMagicIdentifiersCompatible(*calleeDefaultArg, *callerDefaultArg))
         return;
```

Comment:
Is the identifier mismatch logic only for the use of the file identifiers as a default argument value?

Reply:
It's for any magic identifier (including #function, #line, etc.) as a default argument value.